docker-compose.yml

```diff
@@ -1,53 +1,83 @@
 version: '2'
 services:
 
-  postgres:
-    image: postgres
+  database:
+    image: camptocamp/postgis:9.5
+    environment:
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=secret
+    volumes:
+      - postgresql_data:/var/lib/postgresql/data
 
   elasticsearch:
-   image: elasticsearch
+    image: elasticsearch:5.2
+    volumes:
+      - elasticsearch_data:/usr/share/elasticsearch/data
 
   rabbitmq:
-     image: rabbitmq
+     image: rabbitmq:3.6
+     volumes:
+      - rabbitmq_data:/var/lib/rabbitmq
 
   django:
     image: geonode/django
     links:
-      - postgres
+      - database
       - elasticsearch
       - rabbitmq
     command: django-admin.py runserver 0.0.0.0:8000 --settings=geonode.settings
-    env_file:
-      - ./scripts/docker/env/production/django.env
+    ports:
+      - "8000:8000"
+    environment:
+      - SITENAME=GeoNode
+      - SITEURL=http://geonode:8000/
+      - DJANGO_SETTINGS_MODULE=geonode.settings
+      - ALLOWED_HOSTS=['localhost', 'geonode']
+      - C_FORCE_ROOT=1
+      - BROKER_URL=amqp://guest:guest@rabbitmq:5672/
+      - DB_HOST=database
+      - DB_PORT=5432
+      - DB_ADMIN_USER=postgres
+      - DB_ADMIN_PASSWORD=secret
+      - DB_GEONODE_NAME=geonode
+      - DB_GEONODE_USER=geonode
+      - DB_GEONODE_PASSWORD=secret
+      - DB_GEOSERVER_NAME=geoserver
+      - DB_GEOSERVER_USER=geoserver
+      - DB_GEOSERVER_PASSWORD=secret
+    volumes:
```

Comment:
We would first remove the `data_dir_conf` container.
Then create 2 tags for the `geonode/geoserver` image, one `2.9.x-oauth2` and another one named `2.9.x`. These images contain a geoserver datadir coming from http://build.geonode.org/geoserver/latest/data-${GEOSERVER_VERSION}.zip, unzipped into `/mnt/geoserver_datadir`. This directory will be runtime copied in the `geoserver_datadir` named volume if empty (this is standard docker feature).
We'd also add an entrypoint script which updates geoserver configuration according to env vars.